src/store/modules/records/mutations.js

```diff
@@ -0,0 +1,87 @@
+import debug from 'debug'
+import Vue from 'vue'
+
+const logger = debug('app:store:records:mutations')
+const logMutation = (mutationName, ...args) => {
+  logger(`${mutationName} mutation being executed`, ...args)
+}
+
+export default {
+  SET_RECORDS(currentState, { listName, records }) {
+    logMutation('SET_RECORDS', listName, records)
+
+    // Filter out any records that don't have metadata attached to them.
+    //  These are records that were created by other providers.
+    currentState[listName] = records.filter((record) => {
+      return !!record.metadata
+    }) || []
```

Comment:
This fallback is unnecessary, `filter` will always return an array. If `records` is an empty array, `filter` will be as well. However, you should keep this default but move it up to the arguments:

```javascript
SET_RECORDS(currentState, { listName, records = [] }) {
  logMutation('SET_RECORDS', listName, records)

   // Filter out any records that don't have metadata attached to them.
  //  These are records that were created by other providers.
  currentState[listName] = records.filter((record) => {
    return !!record.metadata
  })
},
```

Or return and do nothing if `records` is falsy, whatever makes sense here.